modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/SortProcessorTests.java

```diff
@@ -299,4 +299,25 @@ public void testSortWithTargetField() throws Exception {
         assertEquals(ingestDocument.getFieldValue(targetFieldName, List.class), expectedResult);
     }
 
+    public void testSortWithTargetFieldLeavesOriginalUntouched() throws Exception {
+        IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random());
+        List<Integer> fieldValue = Arrays.asList(1, 5, 4);
+        List<Integer> expectedResult = new ArrayList<>(fieldValue);
+        Collections.sort(expectedResult);
+
+        SortOrder order = randomBoolean() ? SortOrder.ASCENDING : SortOrder.DESCENDING;
+        if (order.equals(SortOrder.DESCENDING)) {
+            Collections.reverse(expectedResult);
+        }
+
+        String fieldName = RandomDocumentPicks.addRandomField(random(), ingestDocument, new ArrayList<>(fieldValue));
+        String targetFieldName;
+        do {
+            targetFieldName = RandomDocumentPicks.randomFieldName(random());
+        } while (fieldName.equals(targetFieldName));
```

Comment:
We really should not have random logic like this anywhere in ES. Do we really need random field names to test whether sorting works? I would rather see explicit fields names "field1" and "field2" and actually test both sort orders than always have these random field names. They have their place, but for things where the field name matters.